utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+            # w.r.t. testing, but it was actually unused.
+
+            # As a special case, Darwin embedded non-simulator platforms have
+            # different behavior for the test condition.
+            if stdlib_target.platform.is_darwin and \
+               stdlib_target.platform.is_embedded and \
+               not stdlib_target.platform.is_simulator:
+                if stdlib_target.platform not in platforms_to_skip_test_host:
+                    test_host_only = True
+                    test = True
+                else:
+                    test = False
+
+            name = stdlib_target.name
+            if build:
+                # FIXME: This logic doesn't make much sense, but
+                # build-script-impl had no comments.
+                if args.build_swift_stdlib_unittest_extra or \
+                       args.validation_test or args.long_test:
+                    swift_stdlib_targets.append("swift-stdlib-"+name)
+                else:
+                    swift_stdlib_targets.append("swift-test-stdlib-"+name)
+            if build_benchmark:
+                swift_benchmark_targets.append("swift-benchmark-"+name)
+                # FIXME: This probably should respect `args.benchmark`, but a
+                # typo (?) in build-script-impl meant we always would do this.
+                swift_run_benchmark_targets.append(
+                    "check-swift-benchmark-"+name)
+            if test:
+                if test_host_only:
+                    suffix = "-non-executable"
+                else:
+                    suffix = ""
+                subset_suffix = ""
+                if args.validation_test and args.long_test:
+                    subset_suffix = "-all"
+                elif args.validation_test:
+                    subset_suffix = "-validation"
+                elif args.long_test:
+                    subset_suffix = "-only_long"
+                else:
+                    subset_suffix = ""
+                swift_test_targets.append("check-swift{}{}-{}".format(
+                    subset_suffix, suffix, name))
+                if args.test_optimized and not test_host_only:
+                    swift_test_targets.append(
+                        "check-swift{}-optimize-{}".format(
+                            subset_suffix, name))
+
+        return {
+            "SWIFT_STDLIB_TARGETS": " ".join(swift_stdlib_targets),
+            "SWIFT_SDKS": " ".join(sorted(swift_sdks)),
+            "SWIFT_BENCHMARK_TARGETS": " ".join(swift_benchmark_targets),
+            "SWIFT_RUN_BENCHMARK_TARGETS": " ".join(
+                swift_run_benchmark_targets),
+            "SWIFT_TEST_TARGETS": " ".join(swift_test_targets),
+        }
+
+    # Add the options for each possible host.
+    options = {
+        args.host_target: get_options_for_target(args.host_target)
+    }
+    for target in (target for options in args.cross_compile_hosts
+                   for target in options):
```

Comment:
What I mean is that:
- When we're talking about a host (e.g. when compiling products like LLVM and looping over `ALL_HOSTS` in `build-script-impl`), we have a variable with `host` in the name (maybe `host_deployment_target` for symmetry), which is of type `DeploymentTarget`
- When we're talking about a target (e.g. when deciding which ones to configure/build/test, or when building Foundation, XCTest or other target libraries), we have a variable called `stdlib_deployment_target` which is also of type `DeploymentTarget`.
- Build isn't really important. It's just another host, with the caveat that we can actually run the stuff it produces (theoretically we could substitute any other host which matches that criteria). In the `build-script-impl` refactoring, it was called `LOCAL_HOST` to reflect that.

This reflects that the specifier is basically just a way for you to communicate a platform to the build-script. The build-script will switch on values of type `DeploymentTarget`, and resolve it to more meaningful types like triples (for hosts) or `SWIFT_SDKS` (for targets). It's fine if we specify hosts and targets using the same type/format; so long as we keep our usage of them clear and unambiguous.